topics/integrate-in-existing-app.md

```diff
@@ -1,242 +1,415 @@
+The credentials are validated and saved to an in-memory database.
 
-Before you begin, learn how to [create and configure a KMM application](create-first-app.md).
+If you aren't familiar with KMM, you can learn how to [create and configure a KMM application from scratch](create-first-app.md) first.
 
-## Decide what to share
+## Prepare an environment for development
 
-An application includes two layers:
+1. Install Android Studio 4.2 or higher and [other tools for KMM development](setup.md) on macOS.
 
-* The **frontend** that defines the user interface and its behavior, including animations and transitions.
-* The **backend** that defines the business logic, data management, network stack, and other components that support the frontend.
+   >You will need a Mac with macOS to complete certain steps in this tutorial, which include writing iOS-specific code and running an iOS application.  
+   >These steps cannot be performed on other operating systems, such as Microsoft Windows. This is due to an Apple requirement.
+   >
+   {type="note"}
 
-KMM presumes that the best user experience is when it is native to the platform itself, and therefore we don’t recommend sharing it in KMM. 
-However, you can share the code for the UI behavior that defines what happens with any user interaction and how the frontend communicates 
-with the backend with the help of compatible architectural patterns.
+2. In Android Studio, create a new project from version control: `https://github.com/Kotlin/kmm-integration-sample`.
 
-On the backend, the logic includes many platform-specific features such as network requests, local database access, 
-hardware manipulation, and cryptographic storage. KMM provides a way to [connect to platform-specific APIs](connect-to-platform-specific-apis.md) 
-with the `expect` declaration, by providing the `actual` implementation for each platform (or using a multiplatform library that does this for you).
+3. Switch to the **Project** view.
 
-To summarize, we recommend sharing the following in your application:
+   ![Project view](project-view-for-integrate.png){width=200}
 
-| Layer | Recommendation on sharing|
-| ----- | -------------- |
-| Business logic | **Yes** |
-| Platform access | **Yes/no**. You’ll still need to [use platform-specific APIs](connect-to-platform-specific-apis.md), but you can share the behavior.|
-| Frontend behavior (reaction to inputs & communication with the backend) | **Yes/no**. Consider these [architectural patterns](#mvp-for-legacy-ui-frameworks).|
-| User interface (including animations & transitions) | **No**. It needs to be platform-specific.|
+## Make your code cross-platform
 
-## Integrate KMM into an existing application
+To make your application work on iOS, you'll first make your code cross-platform, and then you’ll reuse your cross-platform code in a new iOS application.
```

Comment:
minor: To **make** .., you'll first **make** ...
maybe rephrase?

Reply:
good point. Copyeditors missed. Updated.